src/components/NewRoomModal.tsx

```diff
@@ -0,0 +1,167 @@
+		dialogRef.current?.showModal()
+	}, [])
+
+	function handleCreateRoom() {
+		const newRoomNumber = generateAlphaNumeric() + generateAlphaNumeric()
+		setRoomUrl(`${domain}/${newRoomNumber}`)
+		setHostRoomUrl(`${domain}/host/${newRoomNumber}`)
+		if (hostRef.current) hostRef.current.value = ''
+		setHideGoBtn(true)
+		handleShowModal()
+		console.log('%c>>> hostFirstName:', 'color: red', hostFirstName)
+		console.log('%c>>> room:', 'color: #5f0', domain, newRoomNumber)
+	}
+
+	function goToNewRoom() {
+		console.log('%c>>> user room', 'color: red', roomUrl)
+		console.log('%c>>> host room', 'color: yellow', hostRoomUrl)
+		setHideCreateBtn(true)
+		const valuesLocalStorate = JSON.stringify({
+			hostName: hostFirstName,
+			roomUrl: hostRoomUrl,
+		})
+		localStorage.setItem('scrumDivingHostData', valuesLocalStorate)
+		router.push(hostRoomUrl)
+	}
+
+	return (
+		<div className='w-full flex flex-col items-center'>
+			{!isLoading && (
+				<div className='w-full flex flex-col items-center gap-6'>
+					{showNameInput ? (
+						<input
+							type='text'
+							ref={hostRef}
+							placeholder='Enter Your First Name'
+							onChange={handleOnChange}
+							className='input input-bordered input-info w-72 text-gray-200 placeholder:italic placeholder:text-info/90'
+						/>
+					) : (
+						<span className='text-2xl text-gray-300 tracking-wider pb-4'>
+							Welcome back {hostFirstName}!
+						</span>
+					)}
+
+					{/* <p className='text-sm text-gray-400 pt-2'>
+						Then click button to create a new room:
+					</p> */}
+
+					<button
+						type='button'
+						className='btn btn-warning btn-sm w-72 text-base'
+						onClick={handleCreateRoom}
+						disabled={hideCreateBtn}
+					>
+						Click to Create New Room
+					</button>
+				</div>
+			)}
+			<dialog ref={dialogRef} className='modal'>
+				<div className='modal-box pt-8 w-2/3 max-w-[38rem] flex flex-col items-center justify-between gap-12 border border-gray-700'>
+					<h2 className='font-bold text-gray-200 text-[1.6rem]'>
+						{hostFirstName}'s New Room URL
+					</h2>
+					<div className='flex flex-col items-center justify-center'>
+						<span className='w-3/4 text-base text-gray-300 text-center pb-1'>
+							Click the button to copy to the clipboard.
+						</span>
+						<span className='w-3/4 text-base text-gray-300 text-center pb-8'>
+							Share this URL with your team members.
+						</span>
+						<span
+							ref={roomUrlRef}
+							className='text-gray-200 font-mono text-lg pb-6'
+						>
+							{roomUrl}
+						</span>
+
+						<button
+							type='button'
+							onClick={copyToClipboard}
+							className='btn btn-accent btn-sm w-full max-w-72 text-base'
+						>
+							Copy Room URL to Clipboard
+						</button>
+					</div>
+
+					<div className='modal-action w-full gap-3 items-center'>
+						<form method='dialog'>
+							{/* if there is a button in form, it will close the modal */}
+							<button
+								disabled={hideGoBtn}
```

Comment:
todo: This could be a `<Link/>`, no need to be a button, or am I missing something here?

Reply:
What do you mean by `tag`? A tag is `p`, `b`, `h1`, `button`, `a`, etc...

One thing is visual, another thing is semantics. And semantically a `<button>` is different from an `<a>`.  Check this [Links and Buttons Guide - CSS Tricks](https://css-tricks.com/a-complete-guide-to-links-and-buttons/) and [Links vs ButtonsLinks - Eric](https://yatil.net/blog/buttons-vs-links)

Reply:
I meant to say "So it should be an <a> tag? Or in Next.js, a <Link>?"

Okay, those points in the articles make sense. I've actually reworked some things so there isn't a need for the onClick. With the onClick and all the JS in the handler, it needed to be a button.

However, if I use a Link I lose the disabled attribute and have to do something else to get the link to disable. I can make it work, but I'm not happy about it. A button just makes more sense.

Reply:
> "So it should be an <a> tag? Or in Next.js, a <Link>?"

In Next, you use `Link` when pointing to another app route, and `a` for everything else.

Reply:
I believe this one is done and can be resolved.